### DISCUSSION TITLE
[Question] How to configure DCO merged module to install Windows 10 drivers?

### DISCUSSION THREAD

**Discussion by oblomingo** [OpenVPN/ovpn-dco-win#72] [Q&A]
Hey,
I've stuck a little bit with DCO driver installation to WIndows 10 using merged module (*.msm).
I've checked how it configured in example WiX project:
```xml
<Property Id="NETADAPTERCX21">
    <DirectorySearch Id="searchAdvApiDir2" Path="[SystemFolder]" Depth="0">
        <FileSearch Id="searchAdvApiDir2" Name="advapi32.dll" MinVersion="6.3.20348.0"/>
    </DirectorySearch>
</Property>

<Directory Id="TARGETDIR" Name="SourceDir">
    <Directory Id="$(var.PROGRAM_FILES_DIR)">
        <Merge Id="OvpnDcoMergeModule" Language="0" DiskId="1" SourceFile="ovpn-dco-$(var.PRODUCT_PLATFORM).msm">
            <ConfigurationData Name="NETADAPTERCX21_Property" Value="[NETADAPTERCX21]" />
        </Merge>
    </Directory>
</Directory>
```
Here we can see that the installer checks if file exists and pass the NETADAPTERCX21 property to merged module configuration.  Sadly I can't reuse the code, cause I use another installer project. In the module configuration I can pass string variable to the module, but it is not clear what value should be passed to activate Windows 10 driver installation. 
I've tried "false", "0" and empty values and they didn't worked for me. 
My question: what value I should pass to the merge module to install Windows 10 drivers?

**Comment by lstipakov** [comment]
I remember it took me awhile to make configurable merge module. You need to pass a property without value so that Windows 10 driver will be installed. Here is the sample which works for me:

```
<Property Id="LOL" Secure="yes"/>

<Directory Id="TARGETDIR" Name="SourceDir">
    <Directory Id="$(var.PROGRAM_FILES_DIR)">
        <Merge Id="OvpnDcoMergeModule" Language="0" DiskId="1" SourceFile="ovpn-dco-$(var.PRODUCT_PLATFORM).msm">
            <ConfigurationData Name="NETADAPTERCX21_Property" Value="[LOL]" />
        </Merge>
    </Directory>
</Directory>
```

But I don't get `I can't reuse the code, cause I use another installer project`. If you already use MSM, why cannot you use the logic from the sampleinstaller?

**Reply by oblomingo** [reply]
At the moment I use the Advanced Installer project, which has different XML notation.
For example:
```xml
<COMPONENT cid="caphyon.advinst.msicomp.MsiMergeModsComponent">
    <ROW Module="ovpndcoamd64.msm" Path="..\..\ovpn-dco-amd64.msm" Feature="OvpnDcomergemodule" Directory="APPDIR" Params="NETADAPTERCX21_Property=LOL;" Order="1"/>
  </COMPONENT>
```
I've found how to set static value, probably it is possible to attach a property as well. However, Advanced Installer properties should be assigned and have a value (at least a space), in this case, the merge module always get some value and can't install Win10 drivers. It would be nice if the module could accept different values like "0" or "1", probably it would resolve a problem with non-assigned properties in Advanced Installer.

**Comment by lstipakov** [comment]
@oblomingo Looks like there is (still?) an [issue](https://github.com/OpenVPN/openvpn-build/issues/796) with the driver distribution. We are interested in wider adoption of the driver, so maybe we could somehow sort it out together?
